data-structures/ds-pque/java/PriorityQueue.java

```diff
@@ -0,0 +1,65 @@
+public class PriorityQueue {
+    static class Node {
+        int data;
+        int priority;
+        Node next;
+
+    }
+
+    static Node node = new Node();
+
+    static Node newNode(int d, int p) {
+        Node temp = new Node();
+        temp.data = d;
+        temp.priority = p;
+        temp.next = null;
+
+        return temp;
+    }
+
+    static int peek(Node head) {
+        return (head).data;
+    }
+
+    // Remove the element
+    static Node pop(Node head) {
+        Node temp = head;
+        (head) = (head).next;
+        return head;
+    }
+
+    // Add elements according to priority
+    static Node add(Node head, int d, int p) {
+        Node start = (head);
+        Node temp = newNode(d, p);
+        if ((head).priority > p) {
+            temp.next = head;
+            (head) = temp;
+        } else {
+            while (start.next != null && start.next.priority < p) {
+                start = start.next;
+            }
+            temp.next = start.next;
+            start.next = temp;
+        }
+        return head;
+    }
+
+    // Check if list is empty
+    static int isEmpty(Node head) {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6336404)